central/declarativeconfig/manager_impl.go

```diff
@@ -23,6 +30,11 @@ const (
 
 type protoMessagesByType = map[reflect.Type][]proto.Message
 
+// ReconciliationErrorReporter processes declarative resources reconciliation errors.
+type ReconciliationErrorReporter interface {
+	ProcessError(protoValue proto.Message, err error)
+}
+
```

Comment:
Albeit its just a WIP and to be changed - let's move this definition to `singleton.go` for now.